File: factorial_iterative.swift
func factorial(n:Int) -> Int  {
  return factorialIter(1,1,n)
}

func factorialIter(product:Int,counter:Int,maxCount:Int) -> Int {
  if counter > maxCount {
    return product
  } else {
    return factorialIter((product*counter),(counter+1),(maxCount))
  }
}

println(factorial(5))

File: factorial_iterative.py
#!/usr/bin/env python


def factorial(n):
  return factorial_iter(1,1,n)

def factorial_iter(product, counter, max_count):
  if counter > max_count:
    return product
  else:
    return factorial_iter((product*counter),(counter+1), max_count)

print factorial(5)